File: jd-aipay-onboarding/assets/server-examples/nodejs-quickstart/src/query_refund_result_gateway_demo.js
/**
 * AI 付 queryRefundResult 接口 Demo（Node.js 版）。
 */
'use strict';

const {
  buildContent,
  buildHttpHeaders,
  buildSignString,
  encodeBizContent,
  hmacSm3Hex,
  parsePfxBase64,
  postJson,
  tryDecryptResponseBizContent,
} = require('./utils/common');

const ENV = '__ENV__'; // pre | prod | sandbox
const SM2_JD_PUB = '__SM2_JD_PUB__'; // 京东 SM2 公钥证书 Base64 —— 敏感参数，由用户按环境提供

const SECRET_KEY = '__SECRET_KEY__';
const PFX_BASE64 = '__PFX_BASE64__';
const PFX_PASSWORD = '__PFX_PASSWORD__';

const ENDPOINT_URL = '__ENDPOINT_URL__';
const APP_ID = '__APP_ID__';
const AGENT_ID = '__AGENT_ID__';
const MERCHANT_NO = '__MERCHANT_NO__';
const ACQ_MERCHANT_NO = '__ACQ_MERCHANT_NO__';
const ACCESS_TYPE = '__ACCESS_TYPE__'; // 接入类型：SERVICE_MER 服务商 / COMMON 普通商户

// 业务参数
const REFUND_NO = '__REFUND_NO__';

function buildBizJson() {
  const biz = {
    acqMerchantNo: ACQ_MERCHANT_NO, // 收单商户号
    accessType: ACCESS_TYPE, // 接入类型：SERVICE_MER 服务商 / COMMON 普通商户
    refundNo: REFUND_NO, // 退款单号
  };
  return JSON.stringify(biz);
}

async function main() {
  const pfx = parsePfxBase64(PFX_BASE64, PFX_PASSWORD);
  const bizJson = buildBizJson();
  const bizContent = encodeBizContent(bizJson, pfx, SM2_JD_PUB);

  const content = buildContent({
    bizContentEncrypted: bizContent,
    appId: APP_ID,
    agentId: AGENT_ID,
    merchantNo: MERCHANT_NO,
  });
  const signString = buildSignString(content);
  const sign = hmacSm3Hex(signString, SECRET_KEY);
  content.sign = sign;

  const body = JSON.stringify({ data: { content } });
  const headers = buildHttpHeaders(APP_ID);

  console.log('=================== bizContent 明文 ===================');
  console.log(bizJson);
  console.log('=================== 签名原文 ===================');
  console.log(signString);
  console.log('=================== 签名结果 ===================');
  console.log(sign);
  console.log('=================== HTTP Header ===================');
  for (const [k, v] of Object.entries(headers)) {
    console.log(`${k}:${v}`);
  }
  console.log('=================== HTTP Body ===================');
  console.log(body);

  const resp = await postJson(ENDPOINT_URL, headers, body);
  console.log('=================== HTTP Response ===================');
  console.log(resp);

  tryDecryptResponseBizContent(resp, pfx);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: jd-aipay-onboarding/assets/server-examples/python-quickstart/src/aipay_demo/query_refund_result_gateway_demo.py
"""AI 付 queryRefundResult 接口 Demo（Python 版）。"""

from __future__ import annotations

import json
import sys
from collections import OrderedDict
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from aipay_demo import config
from aipay_demo.utils.common import (
    build_content,
    build_http_headers,
    build_sign_string,
    encode_biz_content,
    hmac_sm3_hex,
    parse_pfx_base64,
    post_json,
    try_decrypt_response_biz_content,
)




# 业务参数
REFUND_NO = "__REFUND_NO__"


def build_biz_json() -> str:
    biz = OrderedDict([
        ("acqMerchantNo", config.ACQ_MERCHANT_NO),  # 收单商户号
        ("accessType", config.ACCESS_TYPE),         # 接入类型：SERVICE_MER 服务商 / COMMON 普通商户
        ("refundNo", REFUND_NO),              # 退款单号
    ])
    return json.dumps(biz, ensure_ascii=False, separators=(",", ":"))


def main() -> None:
    pfx = parse_pfx_base64(config.PFX_BASE64, config.PFX_PASSWORD)
    biz_json = build_biz_json()
    biz_content = encode_biz_content(biz_json, pfx, config.SM2_JD_PUB)

    content = build_content(
        biz_content_encrypted=biz_content,
        app_id=config.APP_ID,
        agent_id=config.AGENT_ID,
        merchant_no=config.MERCHANT_NO,
    )
    sign_string = build_sign_string(content)
    sign = hmac_sm3_hex(sign_string, config.SECRET_KEY)
    content["sign"] = sign

    body = json.dumps({"data": {"content": content}}, ensure_ascii=False, separators=(",", ":"))
    headers = build_http_headers(config.APP_ID)

    print("=================== bizContent 明文 ===================")
    print(biz_json)
    print("=================== 签名原文 ===================")
    print(sign_string)
    print("=================== 签名结果 ===================")
    print(sign)
    print("=================== HTTP Header ===================")
    for k, v in headers.items():
        print(f"{k}:{v}")
    print("=================== HTTP Body ===================")
    print(body)

    resp = post_json(config.ENDPOINT_URL, headers, body)
    print("=================== HTTP Response ===================")
    print(resp)

    try_decrypt_response_biz_content(resp, pfx)


if __name__ == "__main__":
    main()
